common/services/user-locations.js

```diff
@@ -0,0 +1,64 @@
+const axios = require('axios')
+
+const { AUTH_PROVIDERS, NOMIS_ELITE2_API } = require('../../config')
+const { decodeAccessToken } = require('../lib/access-token')
+const referenceDataService = require('./reference-data')
+
+function getUserLocations(token) {
+  return _getAgencies(token).then(agencies => {
+    return Promise.all(
+      agencies.map(agencyId =>
+        referenceDataService.getLocations({ nomisAgencyId: agencyId })
```

Comment:
Just looking at this, wonder if this should be in the reference data service as `getLocationByAgencyCode` rather than part of the all locations method. It already contains `getLocationById`.

That method could then be responsible for returning the first result if there are multiple by that agency. Removes that responsibility from this library as it shouldn't care. Might make it easier to test too.

Reply:
In fact, maybe create a `getLocationsByAgencyId` which mimics [`getLocationsById`](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/master/common/services/reference-data.js#L52-L60) and call that from this library?

Reply:
This was my original approach but after our in-person discussion I think we have a way forward.